javascript/javascript1/week4/js1week4.js

```diff
@@ -0,0 +1,133 @@
+            return;
+        }
+    }
+}
+
+
+function greeting(commandRequested) {
+    commandRequested = commandRequested || "";
+    const splittedCommand = commandRequested.split(' ');
+    if (splittedCommand.length >= 4) {
+        //['my,name,is,maha,{karzoun}]
+        //[maha,{karzoun}]
+        const clientName = splittedCommand.splice(3, splittedCommand.length).join(' ');
+
+        if (client.name) {
+            console.log(`welcone back ${client.name} :D
+ `);
+        } else {
+            console.log(`welcone ${clientName} , Thank you for using my program :)`);
+            client.name = clientName;
+        }
+    }
+}
+
+function sayMyName() {
+    if (!client.name) {
+        console.log('You are not registerd yet ! ');
+        return;
+    }
+    console.log(`You are ${client.name}`);
+}
+
+function getToday() {
+    const date = new Date();
+    console.log(`Today is ${date.getDay()} of ${months[date.getMonth()]}`);
+}
+function addTodo(commandRequested) {
+    const todo = extractTodo(commandRequested);
+    client.todo.push(todo);
+}
+
+function removeTodo(commandRequested) {
+    const todo = extractTodo(commandRequested);
+    const todoIndex = client.todo.findIndex(task => task == todo);
+    if (todoIndex == -1) {
+        console.error(`this todo '${todo}' was not found in your list `);
+        return;
+    }
+    client.todo.splice(todoIndex, 1);
+
+}
+
+function extractTodo(commandRequested) {
+    commandRequested = commandRequested || "";
+    const splittedCommand = commandRequested.split(' ');
+    if (splittedCommand.length >= 5) {
+        /// ['add','commmand'.... , 'to','my', 'todo']
+        /// [commmand'.... , 'to','my', 'todo']
+        splittedCommand.splice(0, 1);
+        const todo = splittedCommand.slice(0, splittedCommand.length - 3).join(' ');
+        return todo;
+    }
+}
+
+function setTimer(commandRequested) {
+    commandRequested = commandRequested || "";
+    const splittedCommand = commandRequested.split(' ');
+    /// [set a timer for n unit ]
+    if (splittedCommand.length == 6) {
+        let time = Number.parseInt(splittedCommand[4]);
+        const unit = splittedCommand[5];
+
+        if ("minut".includes(unit)) {
+            time = time * 60;
+        }
+        if ("hour".includes(unit)) {
+            time = time* 60 * 60;
+        }
+
+        setTimeout(() => {
+            console.log('Time is up')
+        }, time * 1000);
+    }
+
+}
+
+
+
+function getCommandIndex(commandRequested) {
+    for (let command of commands) {
+        if (commandRequested.includes(command.pattern))
+            return command.id;
+    }
+    return -1;
+}
+
+let timer;
+let client = { name: null, todo: [] };
+
+const commands = [
+    { id: 0, pattern: "my name is" },
+    { id: 1, pattern: "what is my name" },
+    { id: 2, pattern: "today" },
+    { id: 3, pattern: "add" },
+    { id: 4, pattern: "remove" },
+    { id: 5, pattern: "timer " },
+]
```

Comment:
Nice logic to add `command.pattern` to group the commands. You could also check `startsWith()` method for this task.

Reply:
i will keep it in mind for other time thanks